component/src/test/java/org/wso2/extension/siddhi/io/ethereum/source/TestCaseOfEthereumSource.java

```diff
@@ -0,0 +1,274 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.io.ethereum.source;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.stream.input.source.Source;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.EventPrinter;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestCaseOfEthereumSource {
+
+    private Logger log = Logger.getLogger(TestCaseOfEthereumSource.class);
+    private AtomicInteger eventCount = new AtomicInteger(0);
+    private AtomicBoolean eventArrived = new AtomicBoolean(false);
+    private int waitTime = 3000;
+    private int timeout = 60000;
+
+    @BeforeMethod
+    public void init() {
+
+        eventCount.set(0);
+        eventArrived.set(false);
+    }
+
+    @Test
+    public void ethereumFilterNewBlockTest1() throws InterruptedException {
+
+        log.info("---------------------------------------------------------------------------------------------");
+        log.info("Ethereum Source Test Case of Filtering new blocks.");
+        log.info("---------------------------------------------------------------------------------------------");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String siddhiApp = "@App:name('test') \n" +
+                "@source(type='ethereum', \n" +
+                "@map(type='keyvalue'), \n" +
+                "uri = 'ws://localhost:8546', \n" +
```

Comment:
are you starting ethereum publisher locally by yourself? This test file should be able to run as it is on any machine so if you are starting by yourself please do it programatically inside this test case class